### PR TITLE
Switch intake/arm/elevator control to ProfiledPIDController instead of SmartMotion

### PR DESCRIPTION
**Pull request by balghane**
Some concerns about smart motion including the commentary in [this thread](https://www.chiefdelphi.com/t/understanding-and-tuning-smart-motion-for-an-arm/426639). ProfiledPIDController will definitely work adequately for us. We could also generate motion profiles and do position PID on the sparks, but this is a bit simpler. If it doesn't work we will switch to position PID on-spark, which should be pretty easy since we'll have workable PID values already for position control.

### CODE REVIEW

File: src/main/java/frc/robot/Cal.java [hunk 1]
```diff
@@ -93,18 +89,24 @@ public static final class Intake {
...
-        DEPLOY_ALLOWED_CLOSED_LOOP_ERROR_DEG = 3.0;
+        // Functional not logical
+        DEPLOY_ALLOWED_CLOSED_LOOP_ERROR_DEG = 2.0;
 
-    /** Margin for having achieved the desired intake position (in degrees) */
-    public static final double POSITION_MARGIN_DEGREES = 3.0;
+    /** Input deg, output Volts */
+    public static final double DEPLOY_MOTOR_P = Cal.PLACEHOLDER_DOUBLE,
+        DEPLOY_MOTOR_I = 0.0,
+        DEPLOY_MOTOR_D = Cal.PLACEHOLDER_DOUBLE;
```
Comment (tails618) [inline src/main/java/frc/robot/Cal.java:105]: Why do we know that I is zero, but we don't have guesstimates for P and D?

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -116,14 +116,15 @@ public static final class Lift {
 
     public static final double ELEVATOR_MOTOR_GEAR_RATIO = 14.11;
 
-    public static final double ELEVATOR_RIGHT_ABSOLUTE_ENCODER_RATIO = 26 / 28;
+    public static final double ELEVATOR_RIGHT_ABSOLUTE_ENCODER_RATIO = 26.0 / 28.0;
```
Comment (tails618) [inline src/main/java/frc/robot/Constants.java:119]: Is this necessary, or is it just best practice?

**Review by tails618 — APPROVED** [review]

File: src/main/java/frc/robot/Cal.java (hunk 1 above)
Comment (balghane) [reply src/main/java/frc/robot/Cal.java:105]: This controller type doesn't support iZone so we should try to do this without I for now. With the velocity feedforward I'm guessing we won't need it hopefully

File: src/main/java/frc/robot/Constants.java (hunk 2 above)
Comment (balghane) [reply src/main/java/frc/robot/Constants.java:119]: Bug due to integer division,  26/28 == 0

**Pull request merged**